app/ui/src/test/java/com/fsck/k9/fragment/MessageListAdapterTest.kt

```diff
@@ -0,0 +1,614 @@
+        val fontSizes = FontSizes().apply { messageListSubject = FontSizes.FONT_DEFAULT }
+        val appearance = createMessageListAppearance(fontSizes = fontSizes, senderAboveSubject = true)
+        val adapter = createAdapter(appearance)
+        val cursor = createMessageListCursor()
+
+        val view = adapter.createAndBindView(cursor)
+
+        assertNull(view.secondLineView.getFirstAbsoluteSizeSpanValueOrNull())
+    }
+
+    @Test
+    fun withSenderAboveSubjectAndNonDefaultFontSize_shouldSetTextSizeSpanInSecondLineView() {
+        val fontSizes = FontSizes().apply { messageListSubject = FontSizes.LARGE }
+        val appearance = createMessageListAppearance(fontSizes = fontSizes, senderAboveSubject = true)
+        val adapter = createAdapter(appearance)
+        val cursor = createMessageListCursor()
+
+        val view = adapter.createAndBindView(cursor)
+
+        assertEquals(22, view.secondLineView.getFirstAbsoluteSizeSpanValueOrNull())
+    }
+
+    @Test
+    fun dateWithDefaultFontSize_shouldNotSetTextSizeOfDateView() {
+        val fontSizes = FontSizes().apply { messageListDate = FontSizes.FONT_DEFAULT }
+        val appearance = createMessageListAppearance(fontSizes = fontSizes)
+        val adapter = createAdapter(appearance)
+        val cursor = createMessageListCursor()
+
+        val view = adapter.createAndBindView(cursor)
+
+        assertEquals(DATE_DEFAULT_FONT_SIZE, view.dateView.textSize)
+    }
+
+    @Test
+    fun dateWithNonDefaultFontSize_shouldSetTextSizeOfDateView() {
+        val fontSizes = FontSizes().apply { messageListDate = FontSizes.LARGE }
+        val appearance = createMessageListAppearance(fontSizes = fontSizes)
+        val adapter = createAdapter(appearance)
+        val cursor = createMessageListCursor()
+
+        val view = adapter.createAndBindView(cursor)
+
+        assertEquals(22f, view.dateView.textSize)
+    }
+
+    @Test
+    fun previewWithDefaultFontSize_shouldNotSetTextSizeOfSecondLineView() {
+        val fontSizes = FontSizes().apply { messageListPreview = FontSizes.FONT_DEFAULT }
+        val appearance = createMessageListAppearance(fontSizes = fontSizes, previewLines = 1)
+        val adapter = createAdapter(appearance)
+        val cursor = createMessageListCursor()
+
+        val view = adapter.createAndBindView(cursor)
+
+        assertEquals(SECOND_LINE_DEFAULT_FONT_SIZE, view.secondLineView.textSize)
+    }
+
+    @Test
+    fun previewWithNonDefaultFontSize_shouldSetTextSizeOfSecondLineView() {
+        val fontSizes = FontSizes().apply { messageListPreview = FontSizes.LARGE }
+        val appearance = createMessageListAppearance(fontSizes = fontSizes, previewLines = 1)
+        val adapter = createAdapter(appearance)
+        val cursor = createMessageListCursor()
+
+        val view = adapter.createAndBindView(cursor)
+
+        assertEquals(22f, view.secondLineView.textSize)
+    }
+
+
+    fun createMessageListAppearance(
```

Comment:
This could be pushed to a separate class.

Reply:
What would be the advantage to that?

Reply:
This is technically a dep, which should be a mock if we're doing everything by the book. We're not interested in how deps work inside, we're not interested how they're created too. At least at this level. 

Furthermore, why should a test class have object creation API exposed.  Or even have it at all, private or not.

Keeping this code in here makes very little sense to me. I'd just hide it behind a `MessageListAppearanceFactory::largePreview` `MessageListAppearanceFactory::defaultPreview` along with `FontSizes` class creation.

Reply:
I wouldn't call `MessageListAppearance` a dependency, it's "configuration". The *values* in `MessageListAppearance` influence the behavior of `MessageListAdapter`. Unlike a dependency, there's no behavior inside `MessageListAppearance` (maybe you'll remember that I pushed to make it a simple data class without behavior). Because of that it doesn't really make sense to use a mock instead of the real thing.

Rules are different for test code. Test classes stand on their own, they don't expose a public API for others to use (other than the one used by the test framework). Therefore I avoid using the `private` modifier to cut down on noise.

I don't see the value of extracting this code to another class. Creating a `MessageListAppearance` instance is fundamentally part of creating a `MessageListAdapter` instance. It's not code that would be useful to any other test class.

Reply:
Instead of this 
```
       val fontSizes = FontSizes().apply { messageListPreview = FontSizes.LARGE }
        val appearance = createMessageListAppearance(fontSizes = fontSizes, previewLines = 1)
        val adapter = createAdapter(appearance)
```
That you need to read **and** parse line by line as more or less in GWT as 
> given I create fontSize setting for messageListPreview value Large, and create appearance with that fontSize and preview lines set to 1, when I...

You could have this.
```
 val adapter = createAdapter(MessageListAppearanceFactory.largePreview)
```
Which in GWT would read as

> given I have adapter with largePreview appearance settings, when I bind...`

Further more.
I counted over 30 test methods. Say you cut 2 lines each. It's 60+ LoC less to parse. With factory methods out, you're nearing 100 LoC out of the picture.

Test classes test, factories produce. Why mix those?

Reply:
I agree with the sentiment and tried to make the tests more readable. 

But I like the approach of using builder functions with named arguments better than a separate factory object with properties. Instead of `createAdapter(MessageListAppearanceFactory.noSenderAboveSubjectAndOnePreviewLine)` it's now `createAdapter(senderAboveSubject = false, previewLines = 1)`.

The builders are highly geared towards the tests in `MessageListAdapterTest`. So I think it makes sense to keep them in the same file.